### PR TITLE
fix: temporary workaround for thorchain and cosmos ledger app check missing in hdwallet

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

When attempting to manage accounts for thorchain or cosmos, the app crashes due to missing support for `validateCurrentApp` on these chains. When calling `validateCurrentApp`, the promise is rejected with `Unable to find associated app name for coin: ${coin}` because support for these chains is actually missing from hdwallet.

This issue never presented in web previously because we never attempt to validate the correct app being open on ledger, and instead swallow errors (hence the requirement for the new `throwOnReject` flag added in this PR. 

The workaround here is to bypass this check and allow the user to attempt to proceed, then display an error toast if account fetching fails.

We'll have to follow up with a fix in hdwallet and remove this temporary patch.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

progresses #6806  - will not be closed until hdwallet is updated to support these chains and the patch removed from web.

Low risk

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

Attempt to manage thorchain or cosmos accounts without the corresponding ledger app being open on the device. A toast should appear with an appropriate error message.

Managing an account with the corresponding ledger app open should not show error toast, but should behave normally.

## Screenshots (if applicable)

https://github.com/shapeshift/web/assets/125113430/e7f40b23-607b-445f-8a55-c85bd6f41f21

### CODE REVIEW

File: src/lib/account/account.ts [hunk 1]
```diff
@@ -51,14 +52,19 @@ export const deriveAccountIdsAndMetadata: DeriveAccountIdsAndMetadata = async ar
...
+        throwOnReject,
       }),
     ),
   )
 
   const fulfilledAccountIdsAndMetadata = settledAccountIdsAndMetadata.reduce<AccountMetadataById[]>(
     (acc, result) => {
       if (isRejected(result)) {
-        console.error(result.reason)
+        if (throwOnReject) {
```
Comment (gomesalexandre) [inline src/lib/account/account.ts:63]: 🧠

File: src/assets/translations/en/main.json [hunk 2]
```diff
@@ -1811,6 +1811,12 @@
     "ledgerOpenApp": {
       "title": "Open the %{appName} App",
       "description": "To connect this chain, you will need to open the Ledger app on your device."
+    },
+    "errors": {
+      "accountMetadataFailedToFetch": {
+        "title": "Account Metadata Fetch Failed",
```
Comment (gomesalexandre) [inline src/assets/translations/en/main.json:1817]: This seems like quite a low-level messaging, but given the root cause is eventually going to get fixed, fine with that!

**Review by gomesalexandre — COMMENTED** [review]
Tested locally, confirmed this is happy, though with an issue related to one of the two hard things in software engineering.

Not sure if we want to get this in in this state though however - if this is eventually going to get fixed upstream, it may not worth the effort to fix the caching bug and rather fix the root cause in hdwallet instead? 

- Confirmed it's impossible to currently continue with the wrong app on Ledger ✅ 

image

There's a double toast, although this is eventually going to be removed

- The toast(s) only appear once 🚫 

https://github.com/shapeshift/web/assets/17035424/ee448834-e694-4ac9-8b74-9db92e324c47

- Connecting the correct app is still happy ❓ 

Though on a second try, similarly to the toasts that never appear again, caching means auto-fetching is borked. If hard refreshing the app and trying with the correct app on the first try, this is happy.

https://github.com/shapeshift/web/assets/17035424/6c6aa28a-71ee-4a02-80a7-070f89e29d8c

**Comment by woodenfurniture** [comment]
> Tested locally, confirmed this is happy, though with an issue related to one of the two hard things in software engineering.
> 
> Not sure if we want to get this in in this state though however - if this is eventually going to get fixed upstream, it may not worth the effort to fix the caching bug and rather fix the root cause in hdwallet instead?
> 
> * Confirmed it's impossible to currently continue with the wrong app on Ledger ✅
> 
> image
> There's a double toast, although this is eventually going to be removed
> 
> * The toast(s) only appear once 🚫
> 
>  Screen.Recording.2024-05-08.at.14.02.48.mov 
> * Connecting the correct app is still happy ❓
> 
> Though on a second try, similarly to the toasts that never appear again, caching means auto-fetching is borked. If hard refreshing the app and trying with the correct app on the first try, this is happy.
> 
>  Screen.Recording.2024-05-08.at.14.04.03.mov

Ok, thanks for the testing ser! New changes pushed:
* Clear account paging when opening account import for ledger wallets. This fixes the case where it's possible to view accounts from a previous load without the correct ledger app open on the device.
* Fixed double toast on error.
* Moved toast to top-right for consistency with other toasts in account management.

**Comment by woodenfurniture** [comment]
Root cause fixed in hdwallet, can remove the workaround but need to keep paging reset.

**Comment by gomesalexandre** [comment]
Superseeded by https://github.com/shapeshift/web/pull/6866 with pagination hunks applied there

**Pull request closed without merging**